services/distribution/src/test/java/app/coronawarn/server/services/distribution/assembly/component/BoosterNotificationStructureProviderTest.java

```diff
@@ -0,0 +1,91 @@
+import app.coronawarn.server.services.distribution.dgc.exception.FetchBusinessRulesException;
+import org.junit.Rule;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.rules.TemporaryFolder;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.boot.test.context.ConfigDataApplicationContextInitializer;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.when;
+
+@EnableConfigurationProperties(value = {DistributionServiceConfig.class})
+@ExtendWith(SpringExtension.class)
+@ContextConfiguration(
+    classes = {DigitalGreenCertificateToCborMapping.class,
+        CryptoProvider.class, DistributionServiceConfig.class, ProdDigitalCovidCertificateClient.class,
+        DigitalSigningCertificatesClient.class},
+    initializers = ConfigDataApplicationContextInitializer.class)
+public class BoosterNotificationStructureProviderTest {
+
+  private static final String PARENT_TEST_FOLDER = "parent";
+
+  @Autowired
+  DistributionServiceConfig distributionServiceConfig;
+
+  @Autowired
+  CryptoProvider cryptoProvider;
+
+  @Autowired
+  DigitalGreenCertificateToCborMapping dgcToCborMappingMock;
+
+  @MockBean
+  OutputDirectoryProvider outputDirectoryProvider;
+
+  @MockBean
+  DigitalCovidCertificateClient digitalCovidCertificateClient;
+
+  @MockBean
+  DigitalSigningCertificatesClient digitalSigningCertificatesClient;
+
+  @Rule
+  TemporaryFolder testOutputFolder = new TemporaryFolder();
+
+  @BeforeEach
+  public void setup() throws IOException {
+    // create a specific test folder for later assertions of structures.
+    testOutputFolder.create();
+    File outputDirectory = testOutputFolder.newFolder(PARENT_TEST_FOLDER);
+    Directory<WritableOnDisk> testDirectory = new DirectoryOnDisk(outputDirectory);
+    when(outputDirectoryProvider.getDirectory()).thenReturn(testDirectory);
+  }
+
+  @Test
+  void should_contain_booster_notification_rules() throws FetchBusinessRulesException {
+    when(digitalCovidCertificateClient.getBnRules()).thenThrow(FetchBusinessRulesException.class);
+
+    Writable<WritableOnDisk> boosterNotification = getBoosterNotificationStructure();
+
+    assertThat(boosterNotification.getName()).isEqualTo("booster-notification-rules");
+  }
+
+  private Writable<WritableOnDisk> getBoosterNotificationStructure() {
+    BoosterNotificationStructureProvider underTest = new BoosterNotificationStructureProvider(
```

Comment:
Is there any reason why this is constructed manually?  Usually we can also inject beans here, which makes it easier to see which class is tested